cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResources.java

```diff
@@ -0,0 +1,119 @@
+package com.sequenceiq.cloudbreak.cloud.azure;
+
+import static com.sequenceiq.common.api.type.ResourceType.AZURE_DISK_ENCRYPTION_SET;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.microsoft.azure.management.compute.implementation.DiskEncryptionSetInner;
+import com.sequenceiq.cloudbreak.cloud.EncryptionResources;
+import com.sequenceiq.cloudbreak.cloud.azure.client.AzureClient;
+import com.sequenceiq.cloudbreak.cloud.azure.client.AzureClientService;
+import com.sequenceiq.cloudbreak.cloud.exception.CloudConnectorException;
+import com.sequenceiq.cloudbreak.cloud.model.CloudResource;
+import com.sequenceiq.cloudbreak.cloud.model.Platform;
+import com.sequenceiq.cloudbreak.cloud.model.Variant;
+import com.sequenceiq.cloudbreak.cloud.model.encryption.CreatedDiskEncryptionSet;
+import com.sequenceiq.cloudbreak.cloud.model.encryption.DiskEncryptionSetCreationRequest;
+import com.sequenceiq.cloudbreak.cloud.notification.PersistenceNotifier;
+import com.sequenceiq.common.api.type.CommonStatus;
+
+@Service
+public class AzureEncryptionResources implements EncryptionResources {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AzureEncryptionResources.class);
+
+    private static final Pattern ENCRYPTION_KEY_URL_PATTERN = Pattern.compile("https://([^.]+)\\.vault.*");
+
+    @Inject
+    private AzureClientService azureClientService;
+
+    @Inject
+    private AzureUtils azureUtils;
+
+    @Inject
+    private PersistenceNotifier persistenceNotifier;
+
+    @Override
+    public Platform platform() {
+        return AzureConstants.PLATFORM;
+    }
+
+    @Override
+    public Variant variant() {
+        return AzureConstants.VARIANT;
+    }
+
+    @Override
+    public CreatedDiskEncryptionSet createDiskEncryptionSet(DiskEncryptionSetCreationRequest diskEncryptionSetCreationRequest) {
```

Comment:
This method either returns `CreatedDiskEncryptionSet` or the exception thrown by `getOrCreateDiskEncryptionSetOnCloud` or  `IllegalArgumentException` in case of single RG.